### PR TITLE
[PWX-33952] Automation test for Enter and exit maintenance mode after pool resize

### PR DESCRIPTION
**Pull request by rkinhalkar-px**
Automation test for Enter and exit maintenance mode after pool resize

[PWX-33952] 

[PWX-33952]: https://portworx.atlassian.net/browse/PWX-33952?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Test :- 
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Rushikesh/job/rushi_pxdv-7-dev/121/
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Rushikesh/job/rushi_pxdv-7-dev/122/

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -606,3 +606,63 @@ var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
...
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	stepLog := "should get the existing pool and expand it by initiating a resize-disk and again trigger pool expand on same pool"
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:638]: Change the stepLog description to mention cycle through maintenance mode after resize is complete.

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -606,3 +606,63 @@ var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
 	})
 
 })
+
+var _ = Describe("{PoolExpandResizePoolMaintenanceCycle}", func() {
+
+	var testrailID = 34542842
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/34542842
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:613]: This test case in test rail talks about cycle through maintenance mode after `add-disk` operation. Current test does it after `resize-disk`. Pls point to correct test in testrail.

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -606,3 +606,63 @@ var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
...
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+
+		originalSizeInBytes = poolToResize.TotalSize
+		targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+		targetSizeGiB = targetSizeInBytes / units.GiB
+
+		log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type resize-disk",
+			poolIDToResize, poolToResize.TotalSize/units.GiB, targetSizeGiB)
+		triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK)
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:648]: https://portworx.testrail.net/index.php?/tests/view/34542842 This test talks about cycle through maintenance mode after pool expand with `add-disk` operation. Change the operation type to `api.SdkStoragePool_RESIZE_TYPE_ADD_DISK` and log statements everywhere.

File: tests/basic/pool_function_test.go [hunk 4]
```diff
@@ -606,3 +606,63 @@ var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
...
+		status, err := Inst().V.GetNodeStatus(*storageNode)
+		log.FailOnError(err, fmt.Sprintf("Error getting PX status of node %s", storageNode.Name))
+		dash.VerifyFatal(*status, api.Status_STATUS_MAINTENANCE, fmt.Sprintf("Node %s Status not Online", storageNode.Name))
+
+		// Exit Maintenance Mode
+		err = Inst().V.ExitMaintenance(*storageNode)
+		log.FailOnError(err, fmt.Sprintf("fail to exit node %s in maintenance mode", storageNode.Name))
+		status, err = Inst().V.GetNodeStatus(*storageNode)
+		log.FailOnError(err, fmt.Sprintf("Error getting PX status of node %s", storageNode.Name))
+		dash.VerifyFatal(*status, api.Status_STATUS_OK, fmt.Sprintf("Node %s Status not Online", storageNode.Name))
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:666]: After exiting from maintenance mode, perform addition check that pool size after maintenance mode cycle, still reflects the new size.

**Review by vinayakshnd — CHANGES_REQUESTED** [review]
Have provided some minor comments.

File: tests/basic/pool_function_test.go (hunk 1 above)
Comment (rkinhalkar-px) [reply tests/basic/pool_function_test.go:638]: done

File: tests/basic/pool_function_test.go (hunk 3 above)
Comment (rkinhalkar-px) [reply tests/basic/pool_function_test.go:648]: done

File: tests/basic/pool_function_test.go (hunk 4 above)
Comment (rkinhalkar-px) [reply tests/basic/pool_function_test.go:666]: done

**Comment by rkinhalkar-px** [comment]
> Have provided some minor comments.

made the changes

**Review by vinayakshnd — APPROVED** [review]
LGTM

**Pull request merged**